### PR TITLE
Deprecate legacy qasm2 parser for 0.46

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

The legacy qasm2 parser was remvoed on the main branch already in #11308 for the 1.0.0 release. To ensure that users of the 0.x release series are warned of this API change this commit marks the functionality removed in #11308 as deprecated for the 0.46.0 release.

### Details and comments

### CODE REVIEW

File: releasenotes/notes/deprecate-qasm-legacy-d0b6025493da453a.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    The ``qiskit.converters.ast_to_dag`` function has been deprecated and
+    will be removed in the Qiskit 1.0.0 release. It previously was used to
+    convert the abstract syntax tree generated by the legacy OpenQASM 2 parser
+    (in the :mod:`qiskit.qasm` module which has been deprecated) and convert
+    that directly to a :class:`.DAGCircuit`. As the legacy OpenQASM 2 parser
+    has been deprecated this function will no longer serves a purpose after the
+    legacy parser is removed. If you were previously using this, you can
+    instead parse your OpenQASM 2 files into a :class:`.QuantumCircuit` using
+    the :meth:`.QuantumCircuit.from_qasm_file` or
+    :meth:`.QuantumCircuit.from_qasm_str` constructor methods and then
```
Comment (jakelishman) [inline releasenotes/notes/deprecate-qasm-legacy-d0b6025493da453a.yaml:18]: I may have forgotten about this in the other PR, but weren't we planning to get rid of these and require use of `qasm2.loads` as well?

File: qiskit/converters/ast_to_dag.py [hunk 2]
```diff
@@ -26,8 +26,14 @@
 from qiskit.circuit.barrier import Barrier
 from qiskit.circuit.delay import Delay
 from qiskit.circuit.library import standard_gates as std
+from qiskit.utils.deprecation import deprecate_func
 
 
+@deprecate_func(
+    since="0.46",
+    additional_msg="Instead use the qiskit.qasm2.dump() or qiskit.qasm2.    dumps() function",
```
Comment (jakelishman) [inline qiskit/converters/ast_to_dag.py]: Extra spaces here maybe?

File: releasenotes/notes/deprecate-qasm-legacy-d0b6025493da453a.yaml (hunk 1 above)
Comment (mtreinish) [reply releasenotes/notes/deprecate-qasm-legacy-d0b6025493da453a.yaml:18]: We kept them in the other PR. At least for me the circuit constructor class methods still have value just in that I don't need to remember all the extra arguments to the parser to get it to behave like the legacy parser did (which is the only mode I ever really use for parsing qasm2).

File: qiskit/converters/ast_to_dag.py (hunk 2 above)
Comment (mtreinish) [reply qiskit/converters/ast_to_dag.py]: oh weird, how did this slip in.

```suggestion
    additional_msg="Instead use the qiskit.qasm2.dump() or qiskit.qasm2.dumps() function",
```

File: releasenotes/notes/deprecate-qasm-legacy-d0b6025493da453a.yaml (hunk 1 above)
Comment (jakelishman) [reply releasenotes/notes/deprecate-qasm-legacy-d0b6025493da453a.yaml:18]: That's fair.  I think we ought to remove those methods once we've implemented #10737 and made `qasm2.dumps` match it, at which point `qasm2.loads` will load the output of `qasm2.dumps` without any additional modifications. The arguments in favour of removing `QuantumCircuit.qasm` also imo hold for `from_qasm_str`, it's just the usability concern that you're correctly pointing out in the way, I think.

**Review by jakelishman — APPROVED** [review]

**Pull request merged**